src/plugins/intel_cpu/src/transformations/snippets/aarch64/pass/lowered/adjust_brgemm_copy_b_loop_ports.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "adjust_brgemm_copy_b_loop_ports.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/utils/utils.hpp"
+#include "transformations/snippets/aarch64/op/gemm_copy_b.hpp"
+#include "transformations/snippets/aarch64/op/gemm_cpu.hpp"
+
+namespace ov::intel_cpu {
+
+bool pass::aarch64::AdjustBrgemmCopyBLoopPorts::update_loop_info(
+    const std::shared_ptr<snippets::lowered::UnifiedLoopInfo>& loop_info) {
+    OPENVINO_ASSERT(loop_info, "Invalid loop info pointer");
+    bool modified = false;
+    auto caller = [&](snippets::lowered::LoopPort& loop_port,
+                      snippets::lowered::UnifiedLoopInfo::LoopPortDesc& loop_desc) {
+        const auto& p = *loop_port.get_expr_port();
+        if (p.get_type() == snippets::lowered::ExpressionPort::Input && p.get_index() == 1) {
+            const auto& node = p.get_expr()->get_node();
+            if (auto brg = as_type_ptr<ov::intel_cpu::aarch64::GemmCPU>(node)) {
+                // from format KN to NK64n(64 is n block), and for each K64n, repack to nK8n
+                if (loop_port.is_incremented()) {
+                    if (loop_port.get_dim_idx() == 0) {
+                        // N blocking loop
+                        const auto& in_0_shape = brg->get_input_shape(0);
+                        int64_t K = in_0_shape.back();  // K dimension(K is not blocked)
+                        // NK repacked and padded to to N(K+1)
+                        // ptr_increment is 1, inc is 64. inc=inc*ptr_increment=64*(K+1)
+                        loop_desc.ptr_increment = snippets::utils::dynamic_safe_mul(loop_desc.ptr_increment, (K + 1));
```

Comment:
To sum `K` and `1`, please use `snippets::utils::dynamic_safe_add` - `K` might be dynamic value